vortex-file/src/segments/cache.rs

```diff
@@ -51,8 +51,8 @@ impl InMemorySegmentCache {
 
 #[async_trait]
 impl SegmentCache for InMemorySegmentCache {
-    async fn get(&self, id: SegmentId, alignment: Alignment) -> VortexResult<Option<ByteBuffer>> {
-        Ok(self.0.get(&id).await.map(|b| b.ensure_aligned(alignment)))
```

Comment:
I think removing this check should be fine because we align just before calling `put`. Was ensuring alignment here historically significant?

Reply:
The idea was that if you had a disk cache / network cache you could read into a correctly aligned buffer. But honestly, the `put` call can just extract and persist the alignment if the cache implementation really cares.